server/common/app_config.py

```diff
@@ -242,11 +242,17 @@ def get_client_config(self, data_adaptor):
             "about_legal_privacy": dataset_config.app__about_legal_privacy,
         }
 
-        # dataset_props
+        # corpora dataset_props
         # TODO/Note: putting info from the dataset into the /config is not ideal.
         # However, it is definitely not part of /schema, and we do not have a top-level
         # route for data properties.  Consider creating one at some point.
         corpora_props = data_adaptor.get_corpora_props()
+        if corpora_props and "default_embedding" in corpora_props:
+            default_embedding = corpora_props["default_embedding"]
+            if isinstance(default_embedding, str) and default_embedding.startswith("X_"):
+                default_embedding = default_embedding[2:]  # drop X_ prefix
+            if default_embedding in data_adaptor.get_embedding_names():
```

Comment:
Need a test to verify that `default_embedding` is or is not based on the logic you've outlined with the three conditionals?

Reply:
not entirely sure I understand the question...  But I believe that the tests you are asking about are already done inside `get_embedding_names()`.  It returns a list of the embeddings which are legit - ie, which exist, meet our criteria for serving, etc.

Reply:
Sorry, all I meant was to request modifying `test_app_config.py` to verify that `data_config["config"]["parameters"]["default_embedding"]` gets populated in one of the tests which I don't see today.

Reply:
roger - will do.

Reply:
see test_corpora.py